### PR TITLE
Regenerator improvements

### PR DESCRIPTION
**Pull request by AroneyS**
Change regenerator to take SingleM package as input and output updated SingleM package with sequences and taxonomy replaced by updated versions. GraftM package HMMs are unchanged.

Arguments: 
input_singlem_package, output_singlem_package: 
input_sequences, input_taxonomy: new bacteria/archaea sequences and taxonomy
euk_sequences, euk_taxonomy: candidate eukaryotic sequences and taxonomy (graftM graft used to select matches)
window_position: optionally adjust the window position of the package
sequence_prefix: optionally add a prefix to each sequence ID (e.g. package name)
min_aligned_percent: optionally define GraftM create sequence inclusion parameter

### CODE REVIEW

**Comment by AroneyS** [comment]
I regenerated S2.1 with this but the output S3.1 is not finding the same sequences. I'll do more testing to figure out what is going on.

**Comment by AroneyS** [comment]
Creating and using a regenerated S3 metapackage now works.

**Review by wwood — CHANGES_REQUESTED** [review]
Hey just the tiniest thing. Thanks for this.

File: singlem/package_creator.py [hunk 1]
```diff
@@ -106,13 +122,15 @@ def create(self, **kwargs):
...
                                             gpkg_name, hmm_position, window_size, 
-                                            target_domains, gene_description)
+                                            target_domains, gene_description,
+                                            taxonomy_hash_path)
 
             shutil.rmtree(gpkg_name)
             if is_protein_package:
                 filtered_aligned_tempfile.close()
                 dmnd_tf.close()
+            taxonomy_hash_tf.close()
```
[the remark this answers is not in the document]
Comment (AroneyS) [reply singlem/package_creator.py:134]: Good point

**Pull request merged**